core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala

```diff
@@ -142,4 +143,56 @@ class SnowflakeExpressionBuilder extends SnowflakeParserBaseVisitor[ir.Expressio
     }
   }
 
+  override def visitRanking_windowed_function(ctx: Ranking_windowed_functionContext): ir.Expression = {
+    val windowFunction = buildWindowFunction(ctx)
+    val partitionSpec = buildPartitionSpec(ctx.over_clause())
+    val sortOrder = buildSortOrder(ctx.over_clause())
+    // dummy implementation because the grammar for this is missing
+    val frameSpec = ir.WindowFrame(
+      frame_type = ir.RowsFrame,
+      lower = ir.FrameBoundary(current_row = false, unbounded = true, value = ir.Noop),
+      upper = ir.FrameBoundary(current_row = true, unbounded = false, value = ir.Noop))
+    ir.Window(
+      window_function = windowFunction,
+      partition_spec = partitionSpec,
+      sort_order = sortOrder,
+      frame_spec = frameSpec)
+  }
+
+  private def buildWindowFunction(ctx: Ranking_windowed_functionContext): ir.Expression = {
+    if (ctx.ROW_NUMBER() != null) {
```

Comment:
what do you this about this case as a special case for: https://github.com/databricks/scala-style-guide?tab=readme-ov-file#pattern-matching

<img width="678" alt="image" src="https://github.com/databrickslabs/remorph/assets/259697/bfb765d0-5463-45b1-992e-66dcec2cc07b">

Reply:
btw, found out a statement on early returns - not relevant here, but relevant in other palces:
https://github.com/databricks/scala-style-guide?tab=readme-ov-file#return-statements

<img width="698" alt="image" src="https://github.com/databrickslabs/remorph/assets/259697/d46adb03-8f2f-4007-b2f2-84148b62a5bf">

Reply:
I'm okay with considering if/else if/else chains as pattern-matches and not put curly braces around  going forward.

In fact, as a matter of taste, I prefer to avoid curly braces whenever possible, as an incentive to write most my functions as single expressions. But guidelines are meant precisely to avoid arguing on individual preferences, so...

Reply:
> In fact, as a matter of taste, I prefer to avoid curly braces whenever possible, as an incentive to write most my functions as single expressions. But guidelines are meant precisely to avoid arguing on individual preferences, so...

apparently it aligns with this agreement :)